### PR TITLE
Convert SunPyDeprecationWarnings into errors when running pytest

### PR DESCRIPTION
**Pull request by Cadair**
This should stop us from raising our own deprecation warnings.

### CODE REVIEW

**Comment by nabobalis** [comment]
YOU GONNA HATE ME @Cadair

**Comment by nabobalis** [comment]
I also renamed the functions or replaced them where appropriate. 

You want this done for 0.9?

**Comment by Cadair** [comment]
well that turned into one hell of a purge.

File: sunpy/instr/iris/iris.py [hunk 1]
```diff
@@ -9,9 +9,11 @@
 __all__ = ['SJI_to_cube']
 
 
+#  TODO: rename to SJI_to_sequence? Or delete this?
+
 def SJI_to_cube(filename, start=0, stop=None, hdu=0):
```
Comment (nabobalis) [inline sunpy/instr/iris/iris.py]: We want to rename this?

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -467,22 +466,6 @@ def processing_level(self):
...
-        return u.Quantity([xmin, xmax])
-
-    @property
-    @deprecated("0.8", "This property is only valid for non-rotated WCS")
-    def yrange(self):
-        """Return the Y range of the image from edge to edge."""
-        ymin = self.center.data.lat - self.dimensions[1] / 2. * self.scale[1]
-        ymax = self.center.data.lat + self.dimensions[1] / 2. * self.scale[1]
-        return u.Quantity([ymin, ymax])
-
```
Comment (nabobalis) [inline sunpy/map/mapbase.py]: These are not wanted anymore? For sure?

File: sunpy/map/mapsequence.py [hunk 3]
```diff
@@ -207,8 +207,14 @@ def updatefig(i, im, annotate, ani_data, removes):
...
-                im.set_extent(np.concatenate((ani_data[i].xrange.value,
-                                              ani_data[i].yrange.value)))
+                xmin = ani_data[i].center.data.lon - ani_data[i].dimensions[0] / 2. * ani_data[i].scale[0]
+                xmax = ani_data[i].center.data.lon + ani_data[i].dimensions[0] / 2. * ani_data[i].scale[0]
+                anidata_xrange = u.Quantity([xmin, xmax])
+                ymin = ani_data[i].center.data.lat - ani_data[i].dimensions[1] / 2. * ani_data[i].scale[1]
+                ymax = ani_data[i].center.data.lat + ani_data[i].dimensions[1] / 2. * ani_data[i].scale[1]
+                anidata_yrange = u.Quantity([ymin, ymax])
+                im.set_extent(np.concatenate((anidata_xrange.value,
+                                              anidata_yrange.value)))
```
Comment (nabobalis) [inline sunpy/map/mapsequence.py]: Since `xrange` and `yrange` were deleted, I just copied their code into here since it was the only place it was used.

**Comment by nabobalis** [comment]
The PR patch does seem to reduce the test coverage.

File: sunpy/instr/iris/iris.py (hunk 1 above)
Comment (Cadair) [reply sunpy/instr/iris/iris.py]: I vote rename.

**Comment by Cadair** [comment]
The majority of the coverage decrease appears to be the fact we are no longer running the deprecated decorator :rofl:

File: sunpy/instr/iris/iris.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/instr/iris/iris.py]: Do this now? Or another PR?

**Comment by nabobalis** [comment]
`nabobalis approved these changes on behalf of sunpy/sunpy-maintainers just now`

I CAN DO THAT?

**Comment by nabobalis** [comment]
APPY DONT LIKE IT

**Pull request merged**